orttraining/orttraining/python/training/ortmodule.py

```diff
@@ -436,13 +432,20 @@ def _convert_training_graph_input_to_list(self, *inputs, **kwargs):
         '''
         # User inputs
         non_none_inputs = [inp for inp in inputs if inp is not None]
+        named_buffers_iter = iter(self._original_module.named_buffers())
         result = []
         for input_idx, name in enumerate(self._onnx_graphs_info.user_input_names):
             inp = None
             if input_idx < len(non_none_inputs):
                 inp = non_none_inputs[input_idx]
             elif name in kwargs and kwargs[name] is not None:
                 inp = kwargs[name]
+            elif input_idx >= len(non_none_inputs):
+                # Registered buffers are translated to user_input+initializer in ONNX
+                buffer_name, inp = next(named_buffers_iter)
+                # `name` contains a `_base_module` prefix from `FlattenedOutputModule`
+                assert buffer_name in name, f'Input name {name} expected, but {buffer_name} found!'
```

Comment:
It is also possible here that the user changed the number of inputs from the earlier call to the model to this one. In that scenario, this assert will fail, right? Shall we add a TODO to handle the case when the number of arguments to forward changes from one call to another (in which case, we will have to re-export)?

Reply:
Added a TODO and will make sure UTs for this scenario is pushed on a seParate PR